### PR TITLE
Patternise entity homeostasis traits

### PR DESCRIPTION
**Pull request by rays22**
This PR intends to
1. Remove some homeostasis trait terms from  `entity_attribute_part_of.tsv`.
2. Add these terms to a new entity homeostasis pattern data table.

The goal is to fix the traits that have obsolete GO terms or are at risk to have obsolete GO homeostasis process terms as component terms.

If applied, this commit will fix https://github.com/obophenotype/bio-attribute-ontology/issues/204.

### CODE REVIEW

File: src/patterns/data/default/entity_attribute_location.tsv [hunk 1]
```diff
@@ -248,5 +248,5 @@ OBA:2020005	lysine in blood amount	CHEBI:25094	lysine	PATO:0000070	amount	UBERON
 OBA:2060174		CHEBI:145810	insulin	PATO:0000070	amount	UBERON:0000178	blood
 OBA:2060175		CHEBI:35143	hemoglobin	PATO:0000070	amount	UBERON:0000178	blood
 OBA:2060176		CHEBI:25107	magnesium atom	PATO:0000070	amount	UBERON:0001088	urine
-OBA:2060177		PR:000022172	aspartate aminotransferase	PATO:0000070	amount	UBERON:0000178	blood
+OBA:2060177             PR:000022172    aspartate aminotransferase      PATO:0000070    amount  UBERON:0000178  blood
```
Comment (matentzn) [inline src/patterns/data/default/entity_attribute_location.tsv:251]: Intended change?

File: src/patterns/data/default/entity_attribute_part_of.tsv [hunk 2]
```diff
@@ -3198,3 +3181,4 @@ OBA:1000939		GO:0030916	otic vesicle formation	PATO:0000001	quality
 OBA:1001091	egg shell trait	UBERON:0005079	eggshell	PATO:0000001	quality
 OBA:2020001	life cycle stage trait	UBERON:0000105	life cycle stage	PATO:0000001	quality
 OBA:2020002	plant structure trait	PO:0009011	plant structure	PATO:0000001	quality
+OBA:1000705		GO:0097009	energy homeostasis	PATO:0000001	quality
```
Comment (matentzn) [inline src/patterns/data/default/entity_attribute_part_of.tsv:3184]: Why is this added here, when all the other ones are removed?

File: src/patterns/definitions.owl [hunk 3]
```diff
@@ -50828,797 +50679,27 @@ AnnotationAssertion(Annotation(<http://www.geneontology.org/formats/oboInOwl#has
...
-
-# Class: <http://purl.obolibrary.org/obo/OBA_2060109> (male reproductive organ morphology)
-
-AnnotationAssertion(Annotation(<http://www.geneontology.org/formats/oboInOwl#hasDbXref> "AUTO:patterns/patterns/entity_attribute"^^xsd:string) <http://purl.obolibrary.org/obo/IAO_0000115> <http://purl.obolibrary.org/obo/OBA_2060109> "The morphology of a male reproductive organ."^^xsd:string)
-AnnotationAssertion(Annotation(<http://www.geneontology.org/formats/oboInOwl#hasDbXref> "AUTO:patterns/patterns/entity_attribute"^^xsd:string) <http://www.geneontology.org/formats/oboInOwl#hasExactSynonym> <http://purl.obolibrary.org/obo/OBA_2060109> "morphology of male reproductive organ"^^xsd:string)
-AnnotationAssertion(rdfs:label <http://purl.obolibrary.org/obo/OBA_2060109> "male reproductive organ morphology"^^xsd:string)
-EquivalentClasses(<http://purl.obolibrary.org/obo/OBA_2060109> ObjectIntersectionOf(<http://purl.obolibrary.org/obo/PATO_0000051> ObjectSomeValuesFrom(<http://purl.obolibrary.org/obo/RO_0000052> <http://purl.obolibrary.org/obo/UBERON_0003135>)))
-
-# Class: <http://purl.obolibrary.org/obo/OBA_2060110> (mammillothalamic tract of hypothalamus morphology)
```
Comment (matentzn) [inline src/patterns/definitions.owl:51279]: Why are there more than 1000 lost classes here? This seems wrong!

File: src/patterns/dosdp-patterns/entity_homeostasis_trait.yaml [hunk 4]
```diff
@@ -0,0 +1,58 @@
+---
+pattern_name: entity_homeostasis_trait
+
+pattern_iri: entity_homeostasis_trait.yaml
+
+description: 'Use this pattern for cellular or organism level physiological
+  process traits where the stability of a biologically relevant entity that is
+  internal to an organism or cell is maintained.'
```
Comment (matentzn) [inline src/patterns/dosdp-patterns/entity_homeostasis_trait.yaml:8]: Excellent!

**Review by matentzn — COMMENTED** [review]
I general looks great but check my comments, esp. definitions.owl

File: src/patterns/definitions.owl (hunk 3 above)
Comment (rays22) [reply src/patterns/definitions.owl:51279]: I did remove 118 recently added terms from `entity_attribute.tsv` whose logical axioms look like duplications of terms that already exist in `entity_attribute_part_of.tsv` [in another PR.](https://github.com/obophenotype/bio-attribute-ontology/pull/202/files)

* For example:
    1. existing term in `entity_attribute_part_of.tsv`:
```
1956:OBA:0005653                UBERON:0002990  mammillothalamic tract of hypothalamus  PATO:0000051    morphology
```
    2. intended new term in `entity_attribute.tsv`:

```
4213: OBA:2060110		UBERON:0002990	mammillothalamic tract of hypothalamus	PATO:0000051	morphology
```

I will do some more investigation to figure out of the rest making up the thousand lost classes.

File: src/patterns/data/default/entity_attribute_part_of.tsv (hunk 2 above)
Comment (rays22) [reply src/patterns/data/default/entity_attribute_part_of.tsv:3184]: I had second thoughts about changing the pattern for this one.
Do we want to use [ENVO:2000015	energy](https://www.ebi.ac.uk/ols/ontologies/envo/terms?iri=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FENVO_2000015) as an entity?

Comment (matentzn) [reply src/patterns/data/default/entity_attribute_part_of.tsv:3184]: Sounds wrong to me, but what is meant by the "energy" in "energy homeostasis" - shouldn't this be some role in CHEBI?

File: src/patterns/data/default/entity_attribute_location.tsv (hunk 1 above)
Comment (rays22) [reply src/patterns/data/default/entity_attribute_location.tsv:251]: I had removed it, but after having second thoughts, I have restored this term.
My problem was that `PR:000022172 ` is "A protein that is a translation product of the Escherichia coli K-12 aspC gene or a 1:1 ortholog thereof". Thus it is a foreign protein which is not normally a part of or found in blood. However, I think it sill can be considered a trait, because the new term is independent of the state of the character (i.e. amount = 0 --> normal; amount > 0 abnormal).

File: src/patterns/data/default/entity_attribute_part_of.tsv (hunk 2 above)
Comment (rays22) [reply src/patterns/data/default/entity_attribute_part_of.tsv:3184]: No, there is no 'energy' in CHEBI roles.
The entity in 'energy homeostasis' should mean [ 'energy' as the physical entity **E** that is measured in joules (J)](https://en.wikipedia.org/wiki/Energy).

E = mc²

The ENVO definition looks like the best entity match here for 'energy' (from Ancient Greek: ἐνέργεια).

File: src/patterns/definitions.owl (hunk 3 above)
Comment (rays22) [reply src/patterns/definitions.owl:51279]: I have checked the diff of `definitions.owl`, and the 255 removed classes are all intended to be replaced  in the present PR, obsolete GO terms, or had been already dropped in PR #202 .

File: src/patterns/data/default/entity_attribute_part_of.tsv (hunk 2 above)
Comment (matentzn) [reply src/patterns/data/default/entity_attribute_part_of.tsv:3184]: Ok, Sounds good, will follow your lead!

File: src/patterns/definitions.owl (hunk 3 above)
Comment (matentzn) [reply src/patterns/definitions.owl:51279]: This does not sound right - IDs should never be deleted. At most, we can delete the 120 or so you discovered as duplicates, but not the 255 - these should not have changed their IDs. Even if you move the terms to a new pattern, the ID should be the same, unless I am misunderstanding something here.
